src/commonMain/kotlin/org/kosat/CDCL.kt

```diff
@@ -381,17 +387,17 @@ class CDCL(val clauses: MutableList<MutableList<Int>>, initNumber: Int = 0) {
     }
 
     // add a literal to lemma if it hasn't been added yet
-    private fun updateLemma(lemma: MutableList<Int>, lit: Int) {
+    private fun updateLemma(lemma: Clause, lit: Int) {
```

Comment:
What? Why do you need this method? Just build the (mutable) list of clauses in place and convert it to `Clause` in the end.

Imagine calling this NOT for a lemma. Would the compiler stop you from doing this?